mirdata/datasets/hainsworth.py

```diff
@@ -0,0 +1,223 @@
+    hainsworth with the following structure:
+        > H_1.0/
+            > audio/
+            > annotations/beats
+            > annotations/tempo
+    and copy the hainsworth folder to {}
+    """
+
+
+class Track(core.Track):
+    """Hainsworth dataset class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored. default=None
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Attributes:
+        audio_path (str): path to audio file
+        beats_path (srt): path to beats file
+        tempo_path (srt): path to tempo file
+
+    Cached Properties:
+        beats (BeatData): human-labeled beat annotations
+        tempo (float): human-labeled tempo annotations
+
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        # Audio path
+        self.audio_path = self.get_path("audio")
+
+        # Annotations paths
+        self.beats_path = self.get_path("beats")
+        self.tempo_path = self.get_path("tempo")
+
+    @core.cached_property
+    def beats(self) -> Optional[annotations.BeatData]:
+        return load_beats(self.beats_path)
+
+    @core.cached_property
+    def tempo(self) -> Optional[float]:
+        return load_tempo(self.tempo_path)
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The track's audio
+        Returns:
+           * np.ndarray - audio signal
+           * float - sample rate
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            beat_data=[(self.beats, "beats")],
+            tempo_data=[(self.tempo, "tempo")],
+            metadata=None,
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
+    """Load a Ballroom audio file.
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to an audio file
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+    """
+    return librosa.load(fhandle, sr=None, mono=True)
+
+
+@io.coerce_to_string_io
+def load_beats(fhandle: TextIO):
+    """Load beats
+
+    Args:
+        fhandle (str or file-like): Local path where the beats annotation is stored.
+
+    Returns:
+        BeatData: beat annotations
+
+    """
+    beat_times = []
+    beat_positions = []
+
+    reader = csv.reader(fhandle, delimiter="\t")
+    for line in reader:
```

Comment:
I guess that in Hainsworth all files are properly defined as tsv, not like in Ballroom (https://github.com/mir-dataset-loaders/mirdata/blob/master/mirdata/datasets/ballroom.py#L197)

Reply:
Yes!